c-dependencies/js-compute-runtime/builtins/backend.cpp

```diff
@@ -17,6 +19,617 @@
 #include "host_call.h"
 #include "js/Conversions.h"
 
+enum class Authentication {
+  RSA,
+};
+
+enum class KeyExchange {
+  EECDH,
+  RSA,
+};
+
+enum class Encryption {
+  AES128,
+  AES128GCM,
+  AES256,
+  AES256GCM,
+  CHACHA20POLY1305,
+  TRIPLE_DES,
+};
+
+enum class EncryptionLevel {
+  MEDIUM,
+  HIGH,
+};
+
+enum class MessageDigest {
+  SHA1,
+  SHA256,
+  SHA384,
+  AEAD,
+};
+
+enum class Protocol {
+  SSLv3,
+  TLSv1,
+  TLSv1_2,
+};
+
+class Cipher {
+  std::string openSSLAlias;
+  KeyExchange kx;
+  Authentication au;
+  Encryption enc;
+  MessageDigest mac;
+  Protocol protocol;
+  EncryptionLevel level;
+  int strength_bits;
```

Comment:
This can be a `uint16_t`, instead of `int`, which cuts some more space out of `Cipher`.